packages/api/internal/cache/instance/instance.go

```diff
@@ -63,31 +63,36 @@ func NewCache(analytics analyticscollector.AnalyticsCollectorClient, logger *zap
 
 	cache.OnInsertion(func(ctx context.Context, i *ttlcache.Item[string, InstanceInfo]) {
 		instanceInfo := i.Value()
-		_, err := analytics.InstanceStarted(ctx, &analyticscollector.InstanceStartedEvent{
-			InstanceId:    instanceInfo.Instance.SandboxID,
-			EnvironmentId: instanceInfo.Instance.TemplateID,
-			BuildId:       instanceInfo.BuildID.String(),
-			TeamId:        instanceInfo.TeamID.String(),
-			Timestamp:     timestamppb.Now(),
-		})
-		if err != nil {
-			errMsg := fmt.Errorf("error when sending analytics event: %w", err)
-			telemetry.ReportCriticalError(ctx, errMsg)
+		if analytics != nil {
```

Comment:
If we really need to check these just check them during the initialization and fail the whole initialization.